### PR TITLE
Upgrade deps

### LINKED ISSUE #669: Update JDK and all other deps for v1.4.0

### PR DESCRIPTION
**Pull request by HEdingfield**
Upgrades:
* JDK to 20.0.1
* Gradle to 8.1.1
* spotbugs-gradle-plugin to 5.0.14
* spotbugs to 4.7.3
* checkstyle to 10.12.1
* jlink to 2.26.0
* commons.csv to 1.10.0
* poi-ooxml to 5.2.3
* jackson to 2.15.2
* jupiter to 5.9.3
* javafxplugin to 0.0.14
* javafx to 20.0.1

Closes #669.

### CODE REVIEW

**Comment by artoonie** [comment]
Have you confirmed that the release process (.github/workflows/release.yml) still works? If so, LGTM

**Review by artoonie — APPROVED** [review]

**Comment by HEdingfield** [comment]
> Have you confirmed that the release process (.github/workflows/release.yml) still works? If so, LGTM

Looks like it worked! https://github.com/BrightSpots/rcv/actions/runs/5384588243

**Comment by artoonie** [comment]
Nice! thanks for going through that. :+1:

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -2,11 +2,12 @@ name: "Generate Releases"
...
-#      To test this workflow without creating a release, uncomment the following and add a branch name:
-#      push:
-#        branches:
-#          - 'branch-name'
+    types: [ published ]
+  # To test this workflow without creating a release, uncomment the following and add a branch name (making sure "push" is
+  #  at the same indent level as "release":
+  push:
+    branches:
+      - 'dep-upgrades'
```
Comment (artoonie) [inline .github/workflows/release.yml:10]: make sure to remove before merging

Comment (HEdingfield) [reply .github/workflows/release.yml]: Yep, done!

**Pull request merged**